spec/features/create_etd_spec.rb

```diff
@@ -0,0 +1,235 @@
+    abstract_text = 'this is the abstract text'
+    within '#submissionSteps' do
+      step_list = all('div.step')
+      within step_list[1] do
+        find('div#textareaAbstract').click
+        fill_in 'textareaAbstract_edit', with: abstract_text
+        click_button 'Save'
+      end
+    end
+    # a checked box in the progress section is a background image and has class .progressItemChecked
+    expect(page).to have_content(abstract_text)
+    expect(page.find('#pbAbstractProvided')['style']).to match(/background-image/)
+
+    # the hydra_etd app has all the <input type=file> tags at the bottom of the page, disabled,
+    #   and when uploading files, we have to attach the file to the right one of these elements
+    #   I think this may be an artifact of the js framework it usees, prototype
+    file_upload_elements = all('input[type=file]', visible: false)
+
+    # upload dissertation PDF
+    filename = 'etd_dissertation.pdf'
+    expect(page.find('#pbDissertationUploaded')['style']).to eq '' # dissertation PDF not yet provided
+    expect(page).not_to have_content(filename)
+    dissertation_pdf_upload_input = file_upload_elements.first
+    dissertation_pdf_upload_input.attach_file("spec/fixtures/#{filename}")
+    sleep(3) # wait for upload
+    expect(page).to have_content(filename)
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbDissertationUploaded')['style']).to match(/background-image/)
+
+    # upload supplemental file
+    find('input#cbSupplementalFiles').check
+    filename = 'etd_supplemental.txt'
+    expect(page).not_to have_content(filename)
+    # supplemental files uploaded progress checkbox not visible by default
+    expect(page.find('#pbSupplementalFilesUploaded', visible: false)).not_to be_visible
+    supplemental_upload_input = file_upload_elements[1]
+    supplemental_upload_input.attach_file("spec/fixtures/#{filename}")
+    expect(page).to have_content(filename)
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbSupplementalFilesUploaded')['style']).to match(/background-image/)
+
+    # indicate copyrighted material
+    expect(page.find('#pbPermissionsProvided')['style']).to eq '' # rights not yet selected
+    page.select 'does include', from: 'selectPermissionsOptions'
+
+    # provide copyright permissions letters/files
+    # permission files uploaded progress checkbox not visible by default
+    expect(page.find('#pbPermissionFilesUploaded', visible: false)).not_to be_visible
+    filename = 'etd_permissions.pdf'
+    expect(page).not_to have_content(filename)
+    permissions_upload_input = file_upload_elements[11]
+    permissions_upload_input.attach_file("spec/fixtures/#{filename}")
+    expect(page).to have_content(filename)
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbPermissionsProvided')['style']).to match(/background-image/)
```

Comment:
Can you use have_css selectors?

Reply:
I will check -- checkboxes are wonky because of how they are using javascript and images to show the checkmarks.

Reply:
just FYI - i didn't know how to do this to get the contents of an attribute ...